onnxruntime/core/providers/cuda/nn/dropout.h

```diff
@@ -80,8 +77,7 @@ Status Dropout<trainable_dropout>::ComputeInternal(OpKernelContext* context) con
 
   const Tensor* training_mode = context->Input<Tensor>(2);
   //Check for inference mode.
-  if ((0 == ratio_data /*Backward compat with TrainableDropout*/) ||
-      (!trainable_dropout && (training_mode == nullptr || *(training_mode->Data<bool>()) == false))) {
+  if ((0 == ratio_data) ||(training_mode == nullptr || *(training_mode->Data<bool>()) == false)) {
```

Comment:
>(0 == ratio_data) [](start = 6, length = 17)

same question

Reply:
this also applies to Dropout-12 when training_mode is true and ratio is set to zero (unlikely scenario), in which case we should short-circuit the calculation and take this branch.